### PR TITLE
Support range query

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

Changes I made in this pr:
1. Use promauto to register metrics
2. Add `type` label to `up_custom_query_last_duration`, `up_custom_query_executed_total` and `up_custom_query_errors_total`. The value can be either `query` and `query_range`.
3. Add a `step` flag which defines the default range query step if no step value configured for that query.
4. update go mod go version to 1.14
5. Bump golangci-lint to latest 1.30.0 and not enable all linters by default.
6. Update the queries config file format. Now the file looks like

``` yaml
queries:
- name: this_is_a_range_query
  query: topk(10, count({__name__=~".+"}) by (__name__))
  duration: 1w
  step: 15s
- name: this_is_an_instant_query
  query: up
```

There are two new fields `duration` and `step`. `Duration` is the duration of the range query, if duration is not set, then that query will be considered as an instant query. `Step` is the step value for the range query, if not set then it will use the default step value.

### CODE REVIEW

**Comment by yeya24** [comment]
An example result of running two up instances using range query. One send requests to Thanos querier, one to query frontend.


Screenshot from 2020-08-17 17-39-58

File: pkg/metrics/query.go [hunk 1]
```diff
@@ -64,13 +65,33 @@ func Query(
...
+			Step:  step,
+		})
+		if err != nil {
+			err = fmt.Errorf("querying: %w", err)
+			return warn, err
+		}
+
+		// Don't log response in range query case because there are a lot.
+		level.Debug(l).Log("msg", "request finished", "name", query.Name, "trace-id", rt.TraceID)
+	} else {
```
Comment (kakkoyun) [inline pkg/metrics/query.go:86]: nit: Maybe get rid of else and flatten out with a return.

Comment (yeya24) [reply pkg/metrics/query.go]: Done.

File: cmd/up/main.go [hunk 2]
```diff
@@ -358,14 +376,17 @@ func parseFlags(l log.Logger) (options.Options, error) {
...
+	flag.DurationVar(&opts.Duration, "duration", 5*time.Minute,
 		"The duration of the up command to run until it stops. If 0 it will not stop until the process is terminated.")
 	flag.Float64Var(&opts.SuccessThreshold, "threshold", 0.9, "The percentage of successful requests needed to succeed overall. 0 - 1.")
-	flag.DurationVar(&opts.Latency, "latency", 15*time.Second, //nolint:gomnd
+	flag.DurationVar(&opts.Latency, "latency", 15*time.Second,
 		"The maximum allowable latency between writing and reading.")
-	flag.DurationVar(&opts.InitialQueryDelay, "initial-query-delay", 5*time.Second, //nolint:gomnd
+	flag.DurationVar(&opts.InitialQueryDelay, "initial-query-delay", 5*time.Second,
 		"The time to wait before executing the first query.")
+	flag.DurationVar(&opts.DefaultStep, "step", 30*time.Second, "Default step duration for range query. "+
```
Comment (squat) [inline cmd/up/main.go:387]: Tiny nits:
```suggestion
	flag.DurationVar(&opts.DefaultStep, "step", 30*time.Second, "Default step duration for range queries. "+
```

File: cmd/up/main.go [hunk 3]
```diff
@@ -358,14 +376,17 @@ func parseFlags(l log.Logger) (options.Options, error) {
...
+		"Can be override if step is set in query spec.")
```
Comment (squat) [inline cmd/up/main.go:388]:
```suggestion
		"Can be overridden if step is set in query spec.")
```

**Comment by yeya24** [comment]
> Two small nits

Done.

**Comment by bwplotka** [comment]
:+1:

**Comment by bwplotka** [comment]
The only problem with this is that we run two queries in same time :thinking:

**Pull request merged**